bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs

```diff
@@ -0,0 +1,191 @@
+                    throw new InvalidOperationException("Can't have a revision without script");
+                }
+
+                string revisions = revision ? " (Revisions = true)" : "";
+                if (script)
+                {
+                    SubscriptionCreationOptions subscriptionCreationParams = new SubscriptionCreationOptions
+                    {
+                        Query = "from " + _collectionName + revisions
+                    };
+                    _subscriptionName = await _store.Subscriptions.CreateAsync(subscriptionCreationParams).ConfigureAwait(false);
+                    Console.WriteLine($"Created Subscription with query: '{subscriptionCreationParams.Query}'");
+                }
+                else
+                {
+                    _subscriptionName =  await _store.Subscriptions.CreateAsync<Order>();
+                    Console.WriteLine($"Created Subscription without query");
+                }
+
+                EnableRevisions();
+
+                var workersList = new List<SubscriptionWorker<dynamic>>();
+                var taskList = new List<Task>();
+                Stopwatch sp = null;
+                for (int i = 0; i < workers; i++)
+                {
+                    var worker = _store.Subscriptions.GetSubscriptionWorker(new SubscriptionWorkerOptions(_subscriptionName)
+                    {
+                        Strategy = workers == 1 ? SubscriptionOpeningStrategy.OpenIfFree : SubscriptionOpeningStrategy.Concurrent,
+                        TimeToWaitBeforeConnectionRetry = TimeSpan.FromSeconds(5),
+                        MaxDocsPerBatch = _batchSize
+                    });
+                    workersList.Add(worker);
+                    worker.OnEstablishedSubscriptionConnection += () =>
+                    {
+                        Interlocked.CompareExchange(ref sp, Stopwatch.StartNew(), null);
+                    };
+                }
+
+                var tcs = new TaskCompletionSource<object>();
+                var sentCount = 0;
+                foreach (var worker in workersList)
+                {
+                    taskList.Add(worker.Run((async o =>
+                    {
+                        if (Interlocked.Add(ref sentCount, o.Items.Count) >= _docsAmountToTest)
+                            tcs.TrySetResult(null);
+                        await Task.Delay(fakeProcessingTimePerBatch);
+                    })));
+                }
+
+                await tcs.Task.ConfigureAwait(false);
+
+                foreach (var worker in workersList)
+                {
+                    await worker.DisposeAsync().ConfigureAwait(false);
+                }
+                
+                await Task.WhenAll(taskList);
+                
+                return new RunResult
+                {
+                    DocsProccessed = sentCount,
+                    DocsRequested = _batchSize,
+                    ElapsedMs = sp.ElapsedMilliseconds
```

Comment:
This is wrong. You are also including the time to *shut down*.
Instead, use `TrySetResult(sp.Elapsed)`

Reply:
Done